### ISSUE TITLE
Feature Request - Add data source to determine device information

### ISSUE THREAD

**Issue by tagur87** [jeremmfr/terraform-provider-junos#60]
We would like to be able to retreive device information such as host name, os version, hardware model, and cluster state.

I have investigated this and it should be pretty straightforward to implement a data_source that exposes the data retreived from `<get-system-information/>` and marshalling it into a struct which can be used populate the data source.

```
<get-system-information/>
<rpc-reply xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" xmlns:junos="http://xml.juniper.net/junos/19.3R0/junos">
<system-information>
<hardware-model>srx340</hardware-model>
<os-name>junos-es</os-name>
<os-version>19.3R2-S2.2</os-version>
<serial-number>ABCD123456</serial-number>
<host-name>Test-Hostname</host-name>
<cluster-node/>
</system-information>
```

**Comment by tagur87** [comment]
Can you assign this to me? I already have a proof of concept started, and would like to continue to develop this if you agree.

Thanks!

**Comment by jeremmfr** [comment]
Good idea
I already use `<get-software-information/>` with each new Netconf connection which gives almost the same and a lot of other unnecessary informations and which I store with the current session.  
`<get-system-information/>` seems simpler.

Ok to assign you the creation of this the data_source and the request `<get-system-information/>`
Then, I'll take care of replacing `<get-software-information/>` and simplifying the information collected with each new session.

**Comment by jeremmfr** [comment]
Data source available in release [v1.9.0](https://github.com/jeremmfr/terraform-provider-junos/releases/tag/v1.9.0)
